### DISCUSSION TITLE
API rules for Collection  with :alias and Select Field

### DISCUSSION THREAD

**Discussion by oscar-costa** [pocketbase/pocketbase#4097] [Q&A] [answered]
Hi ganigeorgiev,
I have seen this discussion: https://github.com/pocketbase/pocketbase/discussions/3805#discussioncomment-7634791 , and have tried to take it as a reference to adjust my API rules, but without success.

I have the following collections:
users
establishments
establishments_users: user_id, establishment_id, roles: [standard, content_manager, users_manager]

("establishments_users" as two Relation Fields and one Select Field)

On the Update API rules of "establishments_users" , I'm trying to make some validations:  
Allow update only for registered users
Allow update only if user exists on collection.establishments_users AND as the same establishment_id of the establishments_users record AND as the role "users_manager" asigned

So far, this is the code I have, but i'm not able to check/validate if the user who is updating the record has the "users_manager" role (@collection.establishments_users:auth.roles ?= "users_manager")
```
@request.auth.id != "" // Allow only registered users
// Allow only valid establishment users
&& @collection.establishments_users:auth.user_id ?= @request.auth.id
&& @collection.establishments_users:auth.establishment_id ?= establishment_id.id
&& @collection.establishments_users:auth.roles ?= "users_manager"
```

**Comment by ganigeorgiev** [comment]
I don't think your problem is related to the linked discussion.

`@collection.*` acts as a join and it should be used if you want to reference a collection you don't have direct relation to it (see [Filter syntax](https://pocketbase.io/docs/api-rules-and-filters/#filters-syntax)).

If I'm understanding your description then this doesn't seem to be the case and referencing directly the fields in the "establishments_users" collection API rule should work fine:

```js
@request.auth.id != "" &&
user_id = @request.auth.id &&
// see https://pocketbase.io/docs/api-rules-and-filters/#each-modifier
roles:each ?= "users_manager"
```

**Reply by oscar-costa** [reply]
I ganigeorgiev, the idea is to have a users manager that can update another's user data. They are both related with the same establishment, as they both have been registered in establishments_users collection with the same establishment_id. Only the one with "users_manager" role is permitted to update another user of the same establishment. If not misunderstood, your solution is only for the same user to be able to update himself. Is that correct?

**Reply by ganigeorgiev** [reply]
I'm confused. In your initial description you've written _'On the Update API rules of "establishments_users"'_, so my understanding is that you want an API rule for the "establishments_users" collection. But based on your following comment then I'm no longer sure.

Please elaborate more clearly what is your current collections structure, for what collection you want to define an API rule and what is the desired behavior.

**Reply by ganigeorgiev** [reply]
If you want to apply an API rule for the "users" collection then something like the following should suffice:

```js
@request.auth.id != "" &&
@collection.establishments_users:auth.user_id ?= @request.auth.id &&
@collection.establishments_users:auth.roles:each ?= "users_manager" &&
@collection.establishments_users.user_id ?= id &&
@collection.establishments_users.establishment_id ?= @collection.establishments_users:auth.establishment_id
```

But I'm still not sure that I understand your description so I'll stop engage further to prevent causing further confusion.

**Reply by oscar-costa** [reply]
I send you this structure as a reference, in hope that it is better understood:

collection.users
- user #1
- user #2
- user #...

collection.establishments
- establisment #1
- establisment #2
- establisment #...

collection.establishments_users
- establisments_users #1 { establishment_id: #1, user_id: #1, roles: [content_manager]}
- establisments_users #2 { establishment_id: #1, user_id: #2, roles: [users_manager]}
- establisments_users #... {...}

I want to be able to alow only users that have "users_manager" role defined in collection.establishments_users  (in this example, user #2, through is relation field in establisments_users #2) to be able to update establisments_users #1.

**Reply by oscar-costa** [reply, marked as answer]
Thanks ganigeorgiev. Your last reply helped me to get the correct API Rules for my needs. These are the rules that have worked for me:
```
@request.auth.id != ""
&& @collection.establishments_users:auth.user_id ?= @request.auth.id
&& @collection.establishments_users:auth.establishment_id ?= establishment_id.id
&& @collection.establishments_users:auth.roles:each ?= "users_manager"
```
